plugin/src/App/init.lua

```diff
@@ -138,11 +162,21 @@ function App:render()
 				}, {
 					NotConnectedPage = PluginSettings.with(function(settings)
 						return createPageElement(AppStatus.NotConnected, {
-							onConnect = function(host, port)
-								self:startSession(host, port, {
-									openScriptsExternally = settings:get("openScriptsExternally"),
-									twoWaySync = settings:get("twoWaySync"),
-								})
+							hostRef = self.hostRef,
+							portRef = self.portRef,
+
+							onConnect = function()
+								local hostText = self.hostRef.current.Text
+								local portText = self.portRef.current.Text
+
+								self:startSession(
+									if #hostText > 0 then hostText else Config.defaultHost,
+									if #portText > 0 then portText else Config.defaultPort,
+									{
+										openScriptsExternally = settings:get("openScriptsExternally"),
+										twoWaySync = settings:get("twoWaySync"),
+									}
+								)
```

Comment:
This code is repeated a few times in this file and has gotten more complicated. Can we extract it somewhere else now that the logic has grown?

Reply:
I'm not sure the best way to go about that given that it needs to consume `settings`.

Reply:
We can wrap this component in a function component that pulls out the plugin settings from context and passes them via props. `startSession` can be refactored to pull the values from the refs and settings directly, and we can just call `self:startSession()` in each of these cases.